pkg/controller/compliancesuite/compliancesuite_controller.go

```diff
@@ -502,13 +529,18 @@ func (r *ReconcileComplianceSuite) applyGenericRemediation(rem compv1alpha1.Comp
 // This gets the remediation to be applied. Note that before being able to do that, the machineConfigPool is
 // paused in order to reduce restarts of nodes.
 func (r *ReconcileComplianceSuite) applyMcfgRemediationAndPausePool(rem compv1alpha1.ComplianceRemediation,
+	suite *compv1alpha1.ComplianceSuite,
 	pool *mcfgv1.MachineConfigPool, logger logr.Logger) error {
 	remCopy := rem.DeepCopy()
 	// Only pause pools where the pool wasn't paused before and
 	// the remediation hasn't been applied
 	if !pool.Spec.Paused {
 		logger.Info("Pausing pool", "MachineConfigPool.Name", pool.Name)
 		pool.Spec.Paused = true
+		if remediationNeedsOutdatedRemoval(remCopy, suite) {
```

Comment:
Why is the outdated remediation only touched when unpausing the pool? Wouldn't it mean that only the first outdated MC remediation in a pool gets hits this codepath? in general I think it would be cleaner to have all the code that touches the remediation on one place.

Reply:
That was a mistake from my side. Good catch!